internal/command/launch/launch.go

```diff
@@ -210,25 +200,59 @@ func run(ctx context.Context) (err error) {
 		}
 	}
 
-	if !flag.GetBool(ctx, "generate-name") {
-		if appName := flag.GetString(ctx, "name"); appName == "" {
-			// Prompt the user for the app name
-			if inputName, err := prompt.SelectAppName(ctx); err != nil {
+	if generateName {
+		appConfig.AppName = ""
+	}
+
+	if name != "" {
+		appConfig.AppName = name
+	}
+
+	if !generateName && name == "" {
+		inputName, err := promptForAppName(ctx, appConfig)
+		if err != nil {
+			return err
+		}
+		appConfig.AppName = inputName
+	}
+
+	var org *api.Organization
+	if appConfig.AppName != "" {
+		exists, app, err := appExists(ctx, appConfig)
+		if err != nil {
+			return err
+		}
+
+		if exists {
+			msg := fmt.Sprintf("App %s already exists, do you want to launch into that app?", appConfig.AppName)
+			launchIntoExistingApp, err = prompt.Confirm(ctx, msg)
+			if err != nil {
 				return err
-			} else {
-				appConfig.AppName = inputName
 			}
-		} else {
-			appConfig.AppName = appName
+			if !launchIntoExistingApp {
+				return nil
+			}
+
+			org = &api.Organization{
```

Comment:
hmm, just use `org = app.Organization` ?

Reply:
i spent quite sometime on this actually.  so the type of `app.Organization` is `api.OrganizationBasic`, so I'm unable to assign directly to org, which has a type `api.Organization`.

I had a few options
- use a different query to check if app exists so I can get the organization type. (I didn't take this approach because the other app queries load more information than needed)
- manually create an api.Organization using api.OrganizationBasic (what I did)

I understand how this could cause problems if someone were to use "org" downstream expecting some other fields to exist.

Reply:
sorry I didn't read in time. You were right before.